include/sb_handle/temp_memory_pool.h

```diff
@@ -105,8 +109,19 @@ class Temp_Mem_Pool {
   temp_usm_size_map_t temp_usm_size_map_;
 
   template <typename container_t>
-  void release_buff_mem_(const container_t& mem);
-#endif
+  void release_usm_mem_(const container_t& mem);
+#endif  // SB_ENABLE_USM
 };
 }  // namespace blas
+
+#else
+
+namespace blas {
+// Empty class to serve as a temporary Placeholder asthe feature
+// is not fully supported when using AdaptiveCpp
+class Temp_Mem_Pool;
+}  // namespace blas
```

Comment:
```suggestion
```
Are you sure this is needed? There should be no references to `Temp_Mem_Pool` if `__HIPSYCL__` is defined.

Reply:
Yes  @pgorlani we can remove this but in that case we'll need to hide it conditionally in `portblas_handle.h` with a macro there so that we don't have an undefined attribute type.